thorlcr/master/thgraphmanager.cpp

```diff
@@ -1463,63 +1464,146 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
                         {
-                            WARNLOG("Discarding job with invalid sessionID: wuid=%s, graph=%s (sessionID=%" I64F "d)", currentWuid.str(), currentGraphName.str(), agentSessionID);
-                            currentWuid.clear();
-                        }
-                        else if (querySessionManager().sessionStopped(agentSessionID, 0))
-                        {
-                            WARNLOG("Discarding agentless job: wuid=%s, graph=%s", currentWuid.str(), currentGraphName.str());
+                            WARNLOG("Discarding unknown wuid: wuid=%s, graph=%s", currentWuid.str(), currentGraphName.str());
                             currentWuid.clear();
                         }
                         else
                         {
-                            JobNameScope activeJobName(currentWuid.str());
-                            saveWuidToFile(currentWuid);
-                            PROGLOG("Executing: wuid=%s, graph=%s", currentWuid.str(), currentGraphName.str());
-
+                            SessionId agentSessionID = workunit->getAgentSession();
+                            if (agentSessionID <= 0)
                             {
-                                Owned<IWorkUnit> wu = &workunit->lock();
-                                publishPodNames(wu, currentGraphName, nullptr);
+                                WARNLOG("Discarding job with invalid sessionID: wuid=%s, graph=%s (sessionID=%" I64F "d)", currentWuid.str(), currentGraphName.str(), agentSessionID);
+                                currentWuid.clear();
                             }
-                            SocketEndpoint dummyAgentEp;
-                            jobManager->execute(workunit, currentWuid, currentGraphName, dummyAgentEp);
-
-                            Owned<IWorkUnit> w = &workunit->lock();
-                            if (!multiJobLinger && lingerPeriod)
-                                w->setDebugValue(instance, "1", true);
-
-                            if (jobManager->queryExitException())
+                            else if (querySessionManager().sessionStopped(agentSessionID, 0))
                             {
-                                // NB: exitException has already been relayed.
-                                jobManager->clearExitException();
+                                WARNLOG("Discarding agentless job: wuid=%s, graph=%s", currentWuid.str(), currentGraphName.str());
+                                currentWuid.clear();
                             }
                             else
                             {
-                                switch (w->getState())
+                                SCMStringBuffer jobVersion;
+                                workunit->getDebugValue("platformVersion", jobVersion);
+                                bool mismatch = false;
+                                const char *imagePattern = getenv("imagePattern");
```

Comment:
trivial: could move outside the loop, and also the warning if not set.

Reply:
yes, agree, will move and other related code to pre loop